Tribler/Core/SessionConfig.py

```diff
@@ -730,6 +730,37 @@ def get_watch_folder_path(self):
         return self.sessconfig.get(u'watch_folder', u'watch_folder_dir')
 
     #
+    # API
+    #
+    def set_http_api_enabled(self, http_api_enabled):
+        """
+        Sets whether the HTTP API is enabled.
+        :param http_api_enabled: True or False.
+        """
+        return self.sessconfig.set(u'http_api', u'enabled', http_api_enabled)
+
+    def get_http_api_enabled(self):
+        """
+        Returns whether the HTTP API is enabled.
+        :return: A boolean indicating whether the HTTP API is enabled.
+        """
+        return self.sessconfig.get(u'http_api', u'enabled')
+
+    def set_http_api_port(self, http_api_port):
+        """
+        Sets the HTTP API listen port.
+        :param http_api_port: An integer, indicating the port where the HTTP API should listen on.
+        """
+        return self.sessconfig.set(u'http_api', u'port', http_api_port)
```

Comment:
getRandomPort fetches generates a random number and checks if this port is in use. Suppose now that your http api port is enabled, so you are going to use e.g. port 8081. Now getRandom port checks by chance 8081 and sees that it is not in use, so it will use that one. The moment you want to use 8081 for your http you get an exception.

My idea is to create a list of numbers (socks server, http api port, libtorrent port, etc.) that may be used i.e. they are enabled in the config. getRandomport is then not allowed to take any of these numbers as they may be used.